addons/product/models/product_pricelist.py

```diff
@@ -345,6 +345,11 @@ def get_import_templates(self):
             'label': _('Import Template for Pricelists'),
             'template': '/product/static/xls/product_pricelist.xls'
         }]
+    
+    def unlink(self):
+        if self.env['product.pricelist.item'].search_count([('base', '=', 'pricelist'), ('pricelist_id', 'in', self.ids)]):
+            raise UserError(_('You cannot delete this pricelist, it is used in other pricelist'))
```

Comment:
The .pot files must be updated since you're adding a translated string in a stable version.
Also, you should explain which records are blocking the deletion to explain which records they must delete before this one.